src/services/network-modification-types.ts

```diff
@@ -138,41 +138,41 @@ export interface ShuntCompensatorModificationInfo {
 
 export interface GeneratorModificationInfo {
     studyUuid: string;
-    nodeUuid: UUID;
+    nodeUuid: (string & UUID) | undefined;
     generatorId: string;
-    name: string | null;
-    energySource: string;
-    minP: number;
-    maxP: number;
+    name?: string | null;
+    energySource?: string | null;
+    minP?: number | null;
+    maxP?: number | null;
     ratedS?: number | null;
-    targetP: number | null;
-    targetQ: number | null;
-    voltageRegulation: boolean;
-    targetV: number | null;
-    qPercent: number | null;
+    targetP?: number | null;
+    targetQ?: number | null;
+    voltageRegulation?: boolean | null;
+    targetV?: number | null;
+    qPercent?: number | null;
     voltageLevelId?: string;
     busOrBusbarSectionId?: string;
     isUpdate?: boolean;
     modificationUuid?: string;
-    plannedActivePowerSetPoint: number;
-    marginalCost: number;
-    plannedOutageRate: number;
-    forcedOutageRate: number;
-    directTransX: number;
-    stepUpTransformerX: number;
+    plannedActivePowerSetPoint?: number | null;
+    marginalCost?: number | null;
+    plannedOutageRate?: number | null;
+    forcedOutageRate?: number | null;
+    directTransX?: number | null;
+    stepUpTransformerX?: number | null;
     voltageRegulationType?: any;
-    regulatingTerminalId: string | null;
-    regulatingTerminalType: string | null;
-    regulatingTerminalVlId: string | null;
+    regulatingTerminalId?: string | null;
+    regulatingTerminalType?: string | null;
+    regulatingTerminalVlId?: string | null;
     isReactiveCapabilityCurveOn?: boolean;
-    participate: boolean;
-    droop: number | null;
+    participate?: boolean | null;
+    droop?: number | null;
     maxQ?: number | null;
     minQ?: number | null;
-    reactiveCapabilityCurve?: ReactiveCapabilityCurvePointsData[];
+    reactiveCapabilityCurve?: ReactiveCapabilityCurvePointsData[] | null;
     connectionDirection?: string | null;
     connectionName?: string | null;
-    connectionPosition?: string | null;
+    connectionPosition?: number | null;
     terminalConnected?: boolean | null;
     properties?: Property[];
 }
```

Comment:
to replace by the type in the new file: https://github.com/gridsuite/gridstudy-app/pull/2626/files#diff-a760f582797850710a1a69a0f52c8d8f5153022dcb64318c6f435f4a24228b67R120-R196

keep ? and null for optional fields